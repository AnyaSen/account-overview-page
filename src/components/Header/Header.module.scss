@import "../../assets/scss/variables";

.Header {
  z-index: 999;
  width: 100%;
  background-color: $color-white;
}
