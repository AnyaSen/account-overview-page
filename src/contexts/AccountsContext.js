import React, { useState, createContext } from "react";

export const AccountsContext = createContext();

export const AccountsContextProvider = ({ children }) => {
  const [accNumber, setAccNumber] = useState(90);
  const [accName, setAccName] = useState("FUNDING_ACCOUNT");
  const [accID, setAccID] = useState("DKUW67");
  const [accBalance, setAccBalance] = useState("60â‚¬");

  return (
    <AccountsContext.Provider
      value={{
        accNumber,
        accName,
        accID,
        accBalance
      }}
    >
      {children}
    </AccountsContext.Provider>
  );
};
