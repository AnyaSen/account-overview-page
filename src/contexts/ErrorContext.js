import React, { createContext, useState } from "react";

export const ErrorContext = createContext();

export const ErrorContextProvider = ({ children }) => {
  const [isAccountsError, setIsAccountsError] = useState(false);
  const [isUserInfoError, setIsUserInfoError] = useState(false);

  return (
    <ErrorContext.Provider
      value={{
        isAccountsError,
        setIsAccountsError,
        isUserInfoError,
        setIsUserInfoError
      }}
    >
      {children}
    </ErrorContext.Provider>
  );
};
