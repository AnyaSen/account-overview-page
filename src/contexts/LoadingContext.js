import React, { createContext, useState } from "react";

export const LoadingContext = createContext();

export const LoadingContextProvider = ({ children }) => {
  const [isAccountsLoading, setIsAccountsLoading] = useState(true);
  const [isUserInfoLoading, setIsUserInfoLoading] = useState(true);

  return (
    <LoadingContext.Provider
      value={{
        isAccountsLoading,
        setIsAccountsLoading,
        isUserInfoLoading,
        setIsUserInfoLoading
      }}
    >
      {children}
    </LoadingContext.Provider>
  );
};
