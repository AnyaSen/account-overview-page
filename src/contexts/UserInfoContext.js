import React, { useState, createContext } from "react";

export const UserInfoContext = createContext();

export const UserInfoContextProvider = ({ children }) => {
  const [name, setName] = useState("SOME NAME");

  return (
    <UserInfoContext.Provider
      value={{
        name
      }}
    >
      {children}
    </UserInfoContext.Provider>
  );
};
