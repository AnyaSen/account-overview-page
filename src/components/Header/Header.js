import React, { useContext } from "react";
import Styles from "./Header.module.scss";

import { UserInfoContext } from "../../contexts/UserInfoContext";
import { AccountsContext } from "../../contexts/AccountsContext";

import userPic from "../../assets/img/userPic.svg";

import SearchForm from "../SearchForm/SearchForm";
import UserInfo from "./UserInfo/UserInfo";

export default function Header() {
  const { name } = useContext(UserInfoContext);
  const { accNumber } = useContext(AccountsContext);

  return (
    <div data-testid="Header" className={Styles.Header}>
      <UserInfo
        userName={name}
        userPicture={userPic}
        accountsNumber={accNumber}
      />

      <SearchForm />
    </div>
  );
}
