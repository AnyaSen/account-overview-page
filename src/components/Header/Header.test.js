import React from "react";
import { render, cleanup } from "@testing-library/react";

import { UserInfoContextProvider } from "../../contexts/UserInfoContext";
import { AccountsContextProvider } from "../../contexts/AccountsContext";
import { LoadingContextProvider } from "../../contexts/LoadingContext";
import { ErrorContextProvider } from "../../contexts/ErrorContext";

import Header from "./Header";

describe("<Header />", () => {
  afterAll(cleanup);

  const createRenderTree = () => (
    <ErrorContextProvider>
      <LoadingContextProvider>
        <AccountsContextProvider>
          <UserInfoContextProvider>
            <Header />
          </UserInfoContextProvider>
        </AccountsContextProvider>
      </LoadingContextProvider>
    </ErrorContextProvider>
  );

  test("Should have have 2 children.", () => {
    const tree = createRenderTree();

    const { getByTestId, container } = render(tree);

    const HeaderContainer = getByTestId("Header");

    expect(HeaderContainer.children.length).toEqual(2);

    expect(container.firstChild).toMatchSnapshot();
  });
});
