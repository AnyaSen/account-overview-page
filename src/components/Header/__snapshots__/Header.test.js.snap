// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should have have 2 children. 1`] = `
<div
  class="Header"
  data-testid="Header"
>
  <div
    class="UserInfoContainer"
  >
    <div
      class="UserImgContainer"
    >
      <img
        alt="User Face"
        src="userPic.svg"
      />
      <p>
        SOME NAME
      </p>
    </div>
    <p
      data-testid="number-of-accounts"
    >
      Number of accounts: 
      
    </p>
  </div>
  <div
    class="SearchForm"
    data-testid="Search-form"
  >
    <h1>
      INVESTMENT ACCOUNTS OVERVIEW
    </h1>
    <form
      class="SearchFormContainer"
    >
      <div
        class="inputsContainer"
      >
        <input
          class="InputField"
          data-testid="InputField"
          placeholder="Name"
          type="text"
          value=""
        />
        <input
          class="InputField"
          data-testid="InputField"
          placeholder="ID"
          type="text"
          value=""
        />
      </div>
      <div
        class="btnsContainer"
      >
        <button
          class="PrimaryButton"
          disabled=""
          type="submit"
          value="submit"
        >
          Search
        </button>
        <button
          class="SecondaryButton"
          disabled=""
        >
          Show all
        </button>
      </div>
    </form>
  </div>
</div>
`;
